tests/tests_core/test_experiment.py

```diff
@@ -288,37 +288,41 @@ def test_analyze_statistical_test_with_nan_data(self):
 class OutlierFilteringTestCases(ExperimentTestCase):
     """ Test outlier filtering and quantile filtering. """
 
+    kpi_names = [
+        'normal_same',
+        'normal_shifted',
+        'normal_shifted_by_feature',
+        'normal_unequal_variance'
+    ]
+
     def test_quantile_filtering_multiple_columns(self):
         exp = self.getExperiment()
         flags = exp._quantile_filtering(
             self.data,
-            kpis=[
-                'normal_same',
-                'normal_shifted',
-                'normal_shifted_by_feature',
-                'normal_unequal_variance'
-            ],
-            percentile=99.0,
-            threshold_type='upper'
+            kpis = self.kpi_names,
+            thresholds = { kpi: ('upper', 99.0) for kpi in self.kpi_names }
         )
         self.assertEqual(len(flags[flags==True]), 386)
         filtered = [item[1] for item in list(zip(flags, self.data['variant'])) if item[0] == True]
         filtered_dict = dict((val, filtered.count(val)) for val in set(filtered))
         self.assertEqual(filtered_dict['A'], 88)
         self.assertEqual(filtered_dict['B'], 298)
 
+    def test_outlier_filtering_automatic(self):
+        exp = self.getExperiment()
+        data = exp.outlier_filter(
+            self.data,
+            kpis = [KPI(kpi) for kpi in self.kpi_names],
+        )
+        self.assertEqual(len(self.data) - len(data), exp.metadata['filtered_entities_number'])
+        self.assertEqual(exp.metadata['filtered_entities_number'], 389)
```

Comment:
This test says it removes `389`, but the one about finds `386` to remove. Is there a reason for this difference?

It seems to be the same data, the same KPIs, and the same (default) threshold of 99%. Is it a difference in the threshold_type?

Reply:
Doh! I see my question was already answered

Resolving